river/Seat.zig

```diff
@@ -273,6 +277,8 @@ pub fn setFocusRaw(self: *Self, new_focus: FocusTarget) void {
             PointerConstraint.warpToHint(&self.cursor);
             constraint.sendDeactivated();
             self.cursor.constraint = null;
+        } else {
+            self.cursor.warp();
```

Comment:
Why does this call warp directly instead of setting the flag?

Reply:
here it's for when the output is empty, so it won't trigger an layout event

Reply:
Whether the output is empty or not is irrelevant, focus is double buffered state that is applied through the transaction system.